### PR TITLE
Improve E2E Test for inserting media from library

### LINKED ISSUE #4052: Improve E2E Test for inserting media from library
## Task Description

Following #4039, #4023, let's make sure the existing e2e test for this catches said bug.

https://github.com/google/web-stories-wp/blob/757c3845f5c00e8413126070520b51b3fb70e7eb/tests/e2e/specs/editor/media/insertMediaFromLibrary.js#L27-L44

A percy screenshot would have caught this bug, but the screenshots are skipped because of #1206.

Maybe there are other ways to catch this, too. Like checking for the actual `img` attributes

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Re-enable test and add a new one.

## Relevant Technical Choices

Added a couple of new data selectors to add tests. 

Add new test for inserting from media library. 

## Testing Instructions

---

Fixes #4052

### CODE REVIEW

File: assets/src/edit-story/components/library/panes/media/local/mediaPane.js [hunk 1]
```diff
@@ -223,7 +223,7 @@ function MediaPane(props) {
               options={FILTERS}
               placement={Placement.BOTTOM_START}
             />
-            <Primary onClick={openMediaPicker}>
+            <Primary onClick={openMediaPicker} data-testid="uploadButton">
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/panes/media/local/mediaPane.js]: I don't think there's a need for this; should be able to use `getByRole('button', { name: "Upload" })` in the test

File: tests/e2e/specs/editor/media/insertMediaFromDialog.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    await page.waitForSelector(btnSelector);
+    await page.evaluate((selector) => {
+      document.querySelector(selector).click();
+    }, btnSelector);
+    const btnSelect = '.media-button-select';
+    await page.evaluate((selector) => {
+      document.querySelector(selector).click();
+    }, btnSelect);
+
+    await expect(page).toMatchElement('[data-testid="imageElement"]');
```
Comment (swissspidy) [inline tests/e2e/specs/editor/media/insertMediaFromDialog.js:58]: Can we use the existing `data-testid="frameElement"`? That works for all elements and not just images

File: assets/src/edit-story/elements/image/display.js [hunk 3]
```diff
@@ -100,6 +100,7 @@ function ImageDisplay({ element, box }) {
         src={src}
         srcSet={srcSet}
         alt={resource.alt}
+        data-testid="imageElement"
```
Comment (swissspidy) [inline assets/src/edit-story/elements/image/display.js:103]: As mentioned below, there's probably no need for this as we can re-use the existing `frameElement` test ID

File: assets/src/edit-story/components/library/panes/media/local/mediaPane.js (hunk 1 above)
Comment (spacedmonkey) [reply assets/src/edit-story/components/library/panes/media/local/mediaPane.js]: I don't see why adding this is a problem. We have `data-testid` everywhere.

File: tests/e2e/specs/editor/media/insertMediaFromDialog.js (hunk 2 above)
Comment (spacedmonkey) [reply tests/e2e/specs/editor/media/insertMediaFromDialog.js:58]: Yeah, but the point here is to make sure that the element is an image. I don't want to a selector that works on all elements.

Comment (swissspidy) [reply tests/e2e/specs/editor/media/insertMediaFromDialog.js:58]: Then I'd rather define this test ID in `assets/src/edit-story/components/canvas/frameElement.js`, not in the image's display component.

e.g. changing `data-testid="frameElement"` to 
```
data-testid={`frameElement-${element.type}`}
```

Just need to update `assets/src/edit-story/elements/image/karma/resourceLoading.karma.js` accordingly to use `fixture.screen.getAllByTestId('frameElement-image')`

File: assets/src/edit-story/components/library/panes/media/local/mediaPane.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/components/library/panes/media/local/mediaPane.js]: It's a best practice to use other matchers whenever possible, `data-testid` should be a last resort.

Here `getByRole('button', { name: "Upload" })` should work perfectly fine, so there's no need for a test ID.

Citing the [docs](https://testing-library.com/docs/dom-testing-library/api-queries#bytestid):

> In the spirit of the guiding principles, it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure or styling css class names. Learn more about data-testids from the blog post "Making your UI tests resilient to change"

Comment (spacedmonkey) [reply assets/src/edit-story/components/library/panes/media/local/mediaPane.js]: Isn't `getByRole` part of the react testing library. I didn't think we can use that in puppeteer. I have used something else.

Comment (swissspidy) [reply assets/src/edit-story/components/library/panes/media/local/mediaPane.js]: You're right, I was too deep in looking at some Karma tests while writing this 🤦 

Your current solution looks great!

**Review by swissspidy — APPROVED** [review]

File: tests/e2e/specs/editor/media/insertMediaFromDialog.js (hunk 2 above)
Comment (spacedmonkey) [reply tests/e2e/specs/editor/media/insertMediaFromDialog.js:58]: Changing this is a bigger change than that, frameElement is referenced in karma and e2e test.

**Pull request merged**